client.go

```diff
@@ -87,13 +77,37 @@ func (c *Client) NewSession(ctx context.Context, opts *SessionOptions) (*Session
 		HandleMax:      s.handleMax,
 	}
 	log.Debug(1, "TX (NewSession): %s", begin)
-	_ = s.txFrame(begin, nil)
+
+	// we use send to have positive confirmation on transmission
+	send := make(chan encoding.DeliveryState)
+	_ = s.txFrame(begin, send)
```

Comment:
Should we be ignoring this error? It seems like if we failed to transmit this frame we're going to be in a bad state waiting for a reply that might never come.

Reply:
The error returned here is if the underlying `conn` has been closed, so no bad state.  I agree though that we shouldn't ignore it, fixed.

Reply:
OK this is coming back to me now.  The reason why the error wasn't checked here is because in the following `select` we check `c.conn.Done`, and if `conn` is closed, then we return there.  So, while there's no harm in checking the error at this point, it's essentially duplicating the error check.

Reply:
Thinking about this more, the contract for `txFrame()` is just wonky.  Conceptually it makes sense that it returns an error.  However, in _every single call_ of this method we _never_ check the returned error due to checking `conn.Done`.